app/src/main/java/br/org/cesar/discordtime/stickysessions/ui/lobby/SessionTypeAdapter.java

```diff
@@ -0,0 +1,130 @@
+package br.org.cesar.discordtime.stickysessions.ui.lobby;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.graphics.ColorFilter;
+import android.graphics.PorterDuff;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.util.List;
+
+import androidx.recyclerview.widget.RecyclerView;
+import br.org.cesar.discordtime.stickysessions.R;
+import br.org.cesar.discordtime.stickysessions.domain.model.SessionType;
+import io.reactivex.Observable;
+import io.reactivex.subjects.PublishSubject;
+
+public class SessionTypeAdapter extends RecyclerView.Adapter<SessionTypeAdapter.ViewHolder> {
+
+    private Context mContext;
+    private List<SessionTypeViewModel> mSessionTypes;
+
+    private PublishSubject<SessionType> clickSubject;
+    Observable<SessionType> clickEvent;
+
+    SessionTypeAdapter(Context context, List<SessionTypeViewModel> sessionTypes) {
+        mContext = context;
+        mSessionTypes = sessionTypes;
+        clickSubject = PublishSubject.create();
+        clickEvent = clickSubject;
+    }
+
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View itemView = LayoutInflater
+                .from(parent.getContext())
+                .inflate(R.layout.lobby_card_view, parent, false);
+        return new ViewHolder(itemView);
+    }
+
+    @Override
+    public void onBindViewHolder(ViewHolder holder, int position) {
+        int realPosition = position % mSessionTypes.size();
+
+        SessionTypeViewModel sessionType = mSessionTypes.get(realPosition);
+        holder.media.setImageResource(sessionType.mediaImageRes);
+        holder.title.setText(sessionType.titleRes);
+        holder.description.setText(sessionType.descriptionRes);
+
+        if (sessionType.type == null) {
+            holder.changeToDisabledState();
+        } else {
+            holder.changeToEnabledState();
+        }
+    }
+
+    @Override
+    public int getItemCount() {
```

Comment:
Why are you returning this value here (since you could return sessionTypes.size())?

Reply:
We're returning a big value here to create a cyclic effect on RecyclerView. The sessionTypes.size() is used above to get the right object.